packages/editor/src/components/post-actions/actions.js

```diff
@@ -43,8 +43,8 @@ const trashPostAction = {
 	label: __( 'Move to Trash' ),
 	isPrimary: true,
 	icon: trash,
-	isEligible( { status } ) {
-		return status !== 'trash';
+	isEligible( item, permissions ) {
```

Comment:
What about data views, how would data views pass the permissions object?

What alternative APIs can we offer here?

Reply:
I think we'll have to do the same for DataViews or any other place where the "Post Actions" are used. Gather permission and pass the object.

Reply:
There's another option. Support "thunks" kind of API for isEligible:

```
isEligible: select => item => select( something )
```

I'm not entirely sold on any of the two approaches for the moment.

 - Your approach adds the constraint of having a new "permissions" prop that need to be passed to the DataViews component, and also suffers from the fact that permissions might not be the same for each item.
 - The proposed alternative is too flexible and might leak stuff.

Reply:
> suffers from the fact that permissions might not be the same for each item.

Do you mean that they should be the same for each item in the data views?

Disclaimer: I've little experience with DataView.

Reply:
Depending on the user's role, permissions could be different for each listed item. For example, Authors can edit or delete posts by Admin users.

Passing the `permissions` object isn't a problem; gathering this data with currently available selectors might be.

@spacedmonkey, @TimothyBJacobs, do you have any suggestions on how we can optimally utilize data provided by REST API for DataViews (new `WP_Posts_List_Table`)?

Reply:
Yes! Am I understanding correctly you're asking about how to check things like edit/delete permissions in a list table? Where we don't want to make a separate API request for each item in the list?

The answer is target hints! This is part of the JSON Hyper Schema spec. What the REST API server should be doing is including a `targetHints` property in the link description object for the `self` link. At minimum, it should reference the `allow` header. For example.

I gave a talk about this, unfortunately not recorded but slides are here: https://cdn.timothybjacobs.com/app/media/2018/09/16035712/Schema-Driven-Interfaces.pdf

Two relevant slides:

<img width="1309" alt="image" src="https://github.com/WordPress/gutenberg/assets/3460448/75e12e31-c606-4b62-bc12-c13047f3b45f">

The Core content endpoints don't currently send this information, but it wouldn't be too hard to add if we now need this feature.

Reply:
Definitely something that would be good to add. We'd have to think about how to abstract this properly in core-data as well.

Reply:
Yeah I think we'd be able to make something like a `canUserForEntity` selector that abstracts looking at the `allow` header if present or the target hint.

Reply:
> We'd want to fix #43751 too I think.

Yes, we need entity-aware selectors for permission checks.

I guess we can batch resolve the new selector `canUserForEntity` when `getEntityRecords` is expected. Then, DataViews can call the selector without worrying about extra HTTP requests.

Reply:
@Mamaduka If I'm understanding you're suggesting that the "permissions" object should be part of the "item" object. While for the endpoint API response, that seems logic, I'm not entirely sure about the objects in the state. I feel like these two things are separate concerns and should be in different objects. In the backend you don't have the permissions as part of the Post object for instance.

Reply:
That's not what I mean exactly 😅 

The `_links` are already part of the response object, and the editor relies on them. Example:

https://github.com/WordPress/gutenberg/blob/83faa5a781b4089af53cc5317b5d80c38724d0d8/packages/editor/src/components/post-pending-status/check.js#L16-L17.

Reply:
yes, I think that's wrong :P. I think we should have removed all these metadata from the objects.

Reply:
Obviously, now we can use them if they're already there but would love to actually move away from them.

Reply:
Then, back to the idea board.

Another alternative is to introduce entity aware permission selector (#43751) and use it either by passing permission prop or making `isEligible` a "thunk" like callback.

We could batch resolve this selector for entity lists using the data provided by `targetHints`.